### ISSUE TITLE
Private Transaction compatibility with Ethereum MainNet

### ISSUE THREAD

**Issue by Viserius** [hyperledger/besu#2436]
Hi all. I have been studying Besu's implementation of data privacy and believe I have an idea, or at least a first step, for making data privacy work on the Ethereum mainnet.

I believe that the main reason private transactions are not compatible with the Ethereum MainNet right now, is because of the contents of the private transaction _marker_. A marker has the structure of any "normal" contract execution transaction. However, the payload of a marker is a simple hash and the smart contract address is a flag (such as 0x00000000) for Besu nodes to recognize the transaction as being a marker. 
Usually, these transactions contain a function name and arguments for the payload, and the address of a legitimate smart contract account (SCA).

Thus, as a first step for making it compatible on the Ethereum MainNet, we could do the following.

1. The hash in the payload will be disguised as a normal function execution, i.e. `doNothing(0x2974842...)`. This would allow non-Besu nodes on the mainnet to execute the function (which does nothing), while Besu nodes simply fetch the private tx and execute it like we do now.
2. The "flag" or address of the smart contract will be configurable. Then, we can deploy a smart contract on the MainNet using the do-nothing-function described earlier. The flag will be set to this address.

I do not know if this is a feature you would be interested in, but it could definitely be an interesting option for those who simply want to use the MainNet, rather than maintaining their own network. Obviously, you still need to have private nodes. Simply wanted to express my idea on this possible enhancement.

**Comment by macfarla** [comment]
Hi @Viserius - Besu style private marker transactions are already mainnet compatible. The `to` contract address is not a flag as such, it's a legit address - for the precompile - more info here - https://besu.hyperledger.org/en/stable/Concepts/Privacy/Private-Transaction-Processing/

**Comment by Viserius** [comment]
Hi macfarla, thank you for coming back to me.
I am aware the precompile address has the _format_ of a normal address, but does it also point to an actual Ethereum Smart Contract Account? While looking at the source code, I see that the precompile address [maps to the privacy address](https://github.com/hyperledger/besu/blob/0923de90d60d57b2624ace35eedb096ad3875cb3/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetPrivacyPrecompileAddress.java#L41). This privacy address [maps to Address.PRIVACY](https://github.com/hyperledger/besu/blob/0923de90d60d57b2624ace35eedb096ad3875cb3/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/PrivacyParameters.java#L63) which [is always the constant 126 (or 0x000000000000000000000000000000000000007e)](https://github.com/hyperledger/besu/blob/0923de90d60d57b2624ace35eedb096ad3875cb3/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Address.java#L55). This is not an actual smart contract on the MainNet (Etherscan). Should we not set this option to be configurable, such that regular (non-Besu) nodes on the MainNet see this as a proper message call that can be executed (but simply does not modify the ledger's state, i.e. empty function body)?

**Comment by vmichalik** [comment]
Hi Viserius, I think you might misunderstand how a precompile works - it isn't deployed to a network but exists natively inside the client's EVM implementation. In this case it doesn't affect the public state differently on Besu, it just goes on to make calls to the Private Transaction manager and update the private state. This means it is actually a valid transaction on mainnet (it has also been extensively tested on Ropsten). See the order of operations in Sally's documentation link above for more.

**Comment by Viserius** [comment]
Hi, thank you for coming back to me. I am aware of the precompile, but I was under the impression that you cannot submit a message call with an arbitrary payload structure to a non-SCA address, since non-Besu nodes likely do not have the precompile. After revisiting my test for the Ropsten network, it seems a typo snooped in that made me believe this. After resolving it, I indeed verify that marker transactions can be submitted to non-Besu nodes on the Ropsten network with an arbitrary payload to an arbitrary address. I apologize for the confusion and appreciate the help, thank you! This takes away the necessity of the alternative approach I came up with.
